packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/modeledDataStreamQueryEditor/modeledDataStreamExplorer/modeledDataStreamTable/modeledDataStreamTable.tsx

```diff
@@ -98,19 +98,30 @@ export function ModeledDataStreamTable({
     sorting: {},
   });
 
+  const actionsRef = useRef(actions);
+  const selectedAssetRef = useRef(selectedAsset);
+
+  useEffect(() => {
+    actionsRef.current = actions;
+  }, [actions]);
+
   /**
    * Reset selected items if the user changes the asset
    * to avoid confusing add UX
    */
   useEffect(() => {
-    actions.setSelectedItems([]);
-    /**
-     * adding actions as a dependency causes
-     * the hook to run on every change becuase useCollection
-     * returns a new action reference
-     */
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [selectedAsset]);
+    if (selectedAsset?.assetId !== selectedAssetRef.current?.assetId) {
+      actionsRef.current.setSelectedItems([]);
+      selectedAssetRef.current = selectedAsset;
```

Comment:
Do you need to track the selected asset in a ref? Can you not inspect `collectionProps.selectedItems` to get the information?

Reply:
Yeah, or derive it form the passed in prop. Not sure why I did that.